hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -358,6 +358,14 @@ public boolean hasTargetArtifactAssigned(final long targetId, final String sha1H
                 true);
     }
 
+    @Override
+    public Page<Action> findActiveActionsByTarget(final Pageable pageable, final String controllerId) {
+        if (!actionRepository.activeActionExistsForControllerId(controllerId)) {
```

Comment:
double check if this is needed, in particular the extra DB call

Reply:
This method is only used on the ThingCreated code path, so the likelihood that there are no active actions is rather high than low. The activeActionExistsForControllerId query would detect this case very efficiently. Therefore we should keep it.